CMakeLists.txt

```diff
@@ -52,11 +52,9 @@ OPTION(TAGS				"Generate tags"						OFF)
 OPTION(PROFILE				"Generate profiling information"			OFF)
 OPTION(ENABLE_TRACE			"Enables tracing support"				OFF)
 OPTION(LIBGIT2_FILENAME			"Name of the produced binary"				OFF)
-
-   SET(SHA1_BACKEND 			"CollisionDetection"			       CACHE STRING
-       "Backend to use for SHA1. One of Generic, OpenSSL, Win32, CommonCrypto, mbedTLS, CollisionDetection.")
 OPTION(USE_SSH				"Link with libssh2 to enable SSH support"		 ON)
 OPTION(USE_HTTPS			"Enable HTTPS support. Can be set to a specific backend" ON)
+OPTION(USE_SHA1				"Enable SHA1. Can be set to CollisionDetection(ON)/Backend/Generic" ON)
```

Comment:
Doesn't this mean that a user now cannot choose e.g. OpenSSL for HTTPS and mbedTLS for SHA1? No idea why one would want that, but I've learned that there's always a usecase for the weirdest things. Might be fine to drop that support, though

Reply:
True, I forgot to point that out. The rationale is that backend "cross-breeding" was only supported as a result of sheer luck, since the detection is only made as part of the backend selection. Since are orthogonal, the only "sane" combinations would be the ones listed.